source/common/http/http1/codec_impl.cc

```diff
@@ -376,16 +377,80 @@ void ServerConnectionImpl::onEncodeComplete() {
   }
 }
 
+
+  // handlePath computes the correct :authority and :path headers based on 7230#5.7 and 7230#6
+  //
+  bool
+  ServerConnectionImpl::handlePath(HeaderMapImplPtr& headers, bool is_connect) {
+  // TODO(mattwoodyard) - check configuration option if forward proxy is enabled
+  // In practice there will only ever be one scheme delivered for forward proxy
+  // support
+    HeaderString path(Headers::get().Path);
+
+    // The url is relative nothing to do here
+    if (active_request_->request_url_.c_str()[0] == '/' || active_request_->request_url_.c_str()[0] == '*') {
+      headers->addViaMove(std::move(path), std::move(active_request_->request_url_));
+      return true;
+    }
+
+    if (!codec_settings_.allow_connect_ && !codec_settings_.allow_absolute_url_) {
+      return true;
```

Comment:
Absolute urls definitely did not work. They return 404. This behaviour is preserved unless you supply the allow_absolute_url. In fact there's an explicit check to prevent in https://github.com/lyft/envoy/blob/2095aa0b783925b6ff2ef8b92e0d557af224a7fd/source/common/http/conn_manager_impl.cc#L448 for it.